src/platform/react-hooks/src/hooks/usePresence.ts

```diff
@@ -1,105 +1,62 @@
 import type * as Ably from 'ably';
-import { useCallback, useEffect, useState } from 'react';
+import { useCallback, useEffect, useMemo, useRef, useState } from 'react';
 import { ChannelParameters } from '../AblyReactHooks.js';
 import { useAbly } from './useAbly.js';
-import { useStateErrors } from './useStateErrors.js';
 import { useChannelInstance } from './useChannelInstance.js';
+import { useStateErrors } from './useStateErrors.js';
 
-export interface PresenceResult<T> {
-  presenceData: PresenceMessage<T>[];
+export interface PresenceEnterResult<T> {
   updateStatus: (messageOrPresenceObject: T) => void;
   connectionError: Ably.ErrorInfo | null;
   channelError: Ably.ErrorInfo | null;
 }
 
-export type OnPresenceMessageReceived<T> = (presenceData: PresenceMessage<T>) => void;
-export type UseStatePresenceUpdate = (presenceData: Ably.PresenceMessage[]) => void;
-
 const INACTIVE_CONNECTION_STATES: Ably.ConnectionState[] = ['suspended', 'closing', 'closed', 'failed'];
 
 export function usePresence<T = any>(
   channelNameOrNameAndOptions: ChannelParameters,
   messageOrPresenceObject?: T,
-  onPresenceUpdated?: OnPresenceMessageReceived<T>,
-): PresenceResult<T> {
+): PresenceEnterResult<T> {
   const params =
     typeof channelNameOrNameAndOptions === 'object'
       ? channelNameOrNameAndOptions
       : { channelName: channelNameOrNameAndOptions };
+  const skip = params.skip;
 
   const ably = useAbly(params.ablyId);
   const { channel } = useChannelInstance(params.ablyId, params.channelName);
-
-  const subscribeOnly = typeof channelNameOrNameAndOptions === 'string' ? false : params.subscribeOnly;
-
-  const skip = params.skip;
-
   const { connectionError, channelError } = useStateErrors(params);
 
-  const [presenceData, updatePresenceData] = useState<Array<PresenceMessage<T>>>([]);
-
-  const updatePresence = async (message?: Ably.PresenceMessage) => {
-    const snapshot = await channel.presence.get();
-    updatePresenceData(snapshot);
+  const onMount = useCallback(async () => {
+    await channel.presence.enter(messageOrPresenceObject);
+    // don't add messageOrPresenceObject to dependency list, since it will most likely cause an infinite loop of updates,
+    // in cases when user calls this hook with an object literal instead of a state or memoized object.
+    // it does however prevent us from re-entering presence with new data automatically if it changes.
+    // eslint-disable-next-line react-hooks/exhaustive-deps
```

Comment:
maybe it will be better to put `messageOrPresenceObject` in ref? I don't `// eslint-disable-next-line react-hooks/exhaustive-deps` :) can be dangerous when you change code

Reply:
> // eslint-disable-next-line react-hooks/exhaustive-deps :) can be dangerous when you change code

Yeah, I thought about it too, don't remember why I decided to leave the warning suppression for now (maybe I didn't want to imlement the half-measure? not sure).

Using useRef won't solve the underlying issue - we still won't re-enter if `messageOrPresenceObject` changes.
However, if we introduce any other dependencies in later PRs, `eslint-disable-next-line react-hooks/exhaustive-deps` would suppress those helpful warning.

Will change to `useRef` and move comment over there to note that we don't actually react to `messageOrPresenceObject` changes.

Reply:
Added `useRef` in https://github.com/ably/ably-js/pull/1674/commits/2454a287ceebcad3008b3024509b26c1c1804657